ibis/backends/unix/__init__.py

```diff
@@ -0,0 +1,229 @@
+        del self._tables[name]
+
+    def compile(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Expr, object] | None = None,
+        *,
+        pipe_dir: str | None = None,
+        **_: Any,
+    ):
+        if params is None:
+            params = dict()
+        else:
+            params = {param.op(): value for param, value in params.items()}
+
+        node = expr.as_table().op()
+        node = node.replace(
+            rewrite_join
+            | replace_parameter
+            | bind_unbound_table
+            | split_limit_with_offset,
+            context={"params": params, "backend": self},
+        ).replace(sort_before_join)
+
+        counter = itertools.count()
+
+        commands = []
+
+        def fn(node, _, **kwargs):
+            source = translate(node, **kwargs)
+            # relations are the only nodes that are executed
+            #
+            # value expressions are strings that are passed to the command
+            # (likely awk for any non-trivial computation)
+            if isinstance(node, ops.Relation):
+                path = Path(*filter(None, (pipe_dir, f"t{next(counter)}")))
+                commands.append((source, path))
+                return path
+
+            return source
+
+        node.map(fn)
+        return commands
+
+    def to_pyarrow(self, *args, **kwargs) -> Any:
+        import pyarrow as pa
+
+        return pa.Table.from_pandas(self.execute(*args, **kwargs))
+
+    def execute(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Expr, object] | None = None,
+        **kwargs: Any,
+    ):
+        schema = expr.as_table().schema()
+        columns = schema.names
+        pandas_schema = dict(schema.to_pandas())
+        with (
+            tempfile.TemporaryDirectory() as pipe_dir,
+            concurrent.futures.ThreadPoolExecutor() as exe,
+        ):
+            plan = self.compile(
+                expr, params=params, pipe_dir=pipe_dir, exe=exe, **kwargs
+            )
+
+            for cmd, path in plan:
+                os.mkfifo(path)
+                exe.submit(
+                    lambda cmd, path: Popen(cmd, stdout=path.open(mode="w")),
+                    cmd=cmd,
+                    path=path,
+                )
+
+            # start consuming the output
+            return pd.read_csv(
+                path,
+                header=None,
+                names=columns,
+                dtype=pandas_schema,
+            )
+
+    def explain(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Expr, object] | None = None,
+        **kwargs: Any,
+    ) -> str:
+        plan = self.compile(expr, params=params, **kwargs)
+
+        return "\n".join(
+            f"{shlex.join(list(map(str, cmd)))} > {output.name}" for cmd, output in plan
+        )
+
+    @classmethod
+    def has_operation(self, operation: type[ops.Value]) -> bool:
+        import random
+
+        return random.random() > 0.5
```

Comment:
lol

Reply:
Probabilistic support for all operations!